### PR TITLE
Feat/db sorting for gql

### LINKED ISSUE #863: Querying the transactions GQL endpoint using `from` or `to` attribute should use an index
Create two new indices for the `sender` and `receiver` portion of the transaction. This will look something like the following:

```
fromCF: {send_pk}{global_slot}{txn_hash} -> amount
toCF:   {recv_pk}{global_slot}{txn_hash} -> amount
```

So this allows us to use a prefix iterator on either the sender or receiver and have the transactions sorted by global slot height.

### LINKED ISSUE #873: Querying for all user commands for a public key is too slow
We currently have the following index: `get_commands_for_public_key` but on large public keys there may be 10's of thousands but by default, there is no global slot sorting (aka by wall clock). So for us to achieve correctness on sort ordering, we must process all transactions, then sort and limit; this is too slow.

Proposed solution:

Create two new indices for the `sender` and `receiver` portion of the transaction. This will look something like the following:

```
fromCF: {send_pk}{global_slot}{txn_hash} -> amount
toCF:   {recv_pk}{global_slot}{txn_hash} -> amount
```

So this allows us to use a prefix iterator on either the sender or receiver and have the transactions sorted by global slot height.

### LINKED ISSUE #894: Cannot sort accounts in db by `balance` for GQL queries

### PR DESCRIPTION
**Pull request by Isaac-DeFrain**
There are a lot of changes in this PR all related to storing/sorting data.

- fixes #894
- fixes #863
- fixes #873
- removes unnecessary best ledger clone in `summary` REST endpoint

### CODE REVIEW

File: rust/src/command/mod.rs [hunk 1]
```diff
@@ -142,7 +142,11 @@ pub trait UserCommandWithStatusT {
...
     fn sender(&self) -> PublicKey;
 
+    fn receiver(&self) -> PublicKey;
+
     fn nonce(&self) -> u32;
+
+    fn amount(&self) -> u64;
 }
 
 impl UserCommandWithStatusT for UserCommandWithStatus {
```
Comment (trevorbernard) [inline rust/src/command/mod.rs:152]: UserCommandWithStatusT?

File: rust/src/ledger/public_key.rs [hunk 2]
```diff
@@ -9,13 +9,30 @@ use serde::{Deserialize, Serialize};
 pub struct PublicKey(pub String);
 
 impl PublicKey {
+    pub const LEN: usize = 55;
+    pub const PREFIX: &'static str = "B62q";
```
Comment (trevorbernard) [inline rust/src/ledger/public_key.rs:13]: This seems suspicious. Do we know that this will always be the prefix?

File: rust/src/block/mod.rs [hunk 3]
```diff
@@ -45,15 +45,28 @@ pub struct BlockWithoutHeight {
 pub struct BlockHash(pub String);
 
 impl BlockHash {
-    pub fn from_bytes(bytes: [u8; 32]) -> Self {
-        let block_hash = unsafe { String::from_utf8_unchecked(Vec::from(bytes)) };
-        Self(block_hash)
+    pub const LEN: usize = 52;
+    pub const PREFIX: &'static str = "3N";
```
Comment (trevorbernard) [inline rust/src/block/mod.rs:49]: You don't need to `&'static` prefix if it's const

File: rust/src/block/store.rs [hunk 4]
```diff
@@ -44,6 +47,19 @@ pub trait BlockStore {
     /// Add a block at the given global slot since genesis
     fn add_block_at_slot(&self, state_hash: &BlockHash, slot: u32) -> anyhow::Result<()>;
 
+    /// TODO
```
Comment (trevorbernard) [inline rust/src/block/store.rs:50]: Is this valid?

File: rust/src/block/store.rs [hunk 5]
```diff
@@ -62,4 +78,28 @@ pub trait BlockStore {
...
+    fn set_block_balance_updates(
+        &self,
+        state_hash: &BlockHash,
+        balance_updates: Vec<PaymentDiff>,
+    ) -> anyhow::Result<()>;
+
+    /// TODO
+    fn get_block_balance_updates(&self, state_hash: &BlockHash)
+        -> anyhow::Result<Vec<PaymentDiff>>;
```
Comment (trevorbernard) [inline rust/src/block/store.rs:101]: What's with all the TODOs?

**Comment by trevorbernard** [comment]
There is way too much going on in this PR IMHO but good work

File: rust/src/command/mod.rs (hunk 1 above)
Comment (Isaac-DeFrain) [reply rust/src/command/mod.rs:152]: Trait for `UserCommand` functions. It was added with v2 blocks.

File: rust/src/ledger/public_key.rs (hunk 2 above)
Comment (Isaac-DeFrain) [reply rust/src/ledger/public_key.rs:13]: We do. We've been using them for a long time now. Just wanted to make them a little handier.

File: rust/src/block/store.rs (hunk 5 above)
Comment (Isaac-DeFrain) [reply rust/src/block/store.rs:101]: Draft. Will leave comments

File: rust/src/block/store.rs (hunk 4 above)
Comment (Isaac-DeFrain) [reply rust/src/block/store.rs:50]: 💯

File: rust/src/block/store.rs (hunk 5 above)
Comment (Isaac-DeFrain) [reply rust/src/block/store.rs:101]: Just wait until you hear what that `common_ancestor_account_balance_updates` is for!

File: rust/src/block/mod.rs (hunk 3 above)
Comment (Isaac-DeFrain) [reply rust/src/block/mod.rs:49]: image

**Review by jhult — APPROVED** [review]
Well done, sir!

**Pull request merged**